src/components/CommentItem.jsx

```diff
@@ -1,9 +1,48 @@
-function CommentItem({ comment }) {
+import { useContext, useState } from "react";
+import { UserContext } from "../contexts/User";
+import { deleteComment } from "../api/api";
+
+function CommentItem({ comment, setCommentHasBeenDeleted }) {
+  const { user } = useContext(UserContext);
+  const [isError, setIsError] = useState(false);
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  function handleDelete() {
+    setIsError(false);
+    setIsDeleting(true);
+    setCommentHasBeenDeleted(false);
+    deleteComment(comment.comment_id)
+      .then(() => {
+        setIsDeleting(false);
+        setCommentHasBeenDeleted(comment.comment_id);
+      })
+      .catch((error) => {
+        setIsError(true);
+        setIsDeleting(false);
+      });
```

Comment:
Great implementation of comment deletion, love that you're updating the user on the progress of comment deletion and also if there is an error preventing it 🙌